cmd/bundle.go

```diff
@@ -239,3 +261,95 @@ func showBundleInfo(bundleName string, registryName string) {
+func stampBundleMetadata(bundleMetaFilename string, containerMetaFilename string) {
+	workingDir, err := os.Getwd()
+	if err != nil {
+		log.Errorf("Couldn't determine working directory")
+		return
+	}
+
+	bundleMetaPath := filepath.Join(workingDir, bundleMetaFilename)
+	containerMetaPath := filepath.Join(workingDir, containerMetaFilename)
+
+	bundleMeta, errB := ioutil.ReadFile(bundleMetaPath)
+	if errB != nil {
+		log.Errorf("Bundle metadata file [%s] not found in working directory", bundleMetaFilename)
+	}
+	containerMeta, errC := ioutil.ReadFile(containerMetaPath)
+	if errC != nil {
+		log.Errorf("Container metadata file [%s] not found in working directory", containerMetaFilename)
+	}
+	if errB != nil || errC != nil {
+		return
+	}
+
+	newContainerMeta := addBundleMetadata(bundleMeta, containerMeta)
+	if len(newContainerMeta) == 0 {
+		log.Errorf("Failed to add [%s] contents to [%s]", bundleMetaFilename, containerMetaFilename)
+		return
+	}
+
+	writeErr := ioutil.WriteFile(containerMetaFilename, newContainerMeta, 0644)
+	if writeErr != nil {
+		log.Errorf("Container metadata file [%s] could not be written", containerMetaFilename)
+		return
+	}
+	fmt.Printf("Wrote b64 encoded [%s] to [%s]\n", bundleMetaFilename, containerMetaFilename)
+}
+
+func addBundleMetadata(bundleMeta []byte, containerMeta []byte) []byte {
+	lineBreakAfter := 76
+	lineBreakText := []byte("\\\n")
+
+	// Encode bundle metadata as b64 and add linebreaks
+	bundleMetaEncoded := make([]byte, base64.StdEncoding.EncodedLen(len(bundleMeta)))
+	base64.StdEncoding.Encode(bundleMetaEncoded, bundleMeta)
+	bundleMetaEncoded = addLineBreaks(bundleMetaEncoded, lineBreakText, lineBreakAfter)
+
+	// Match the apb spec "LABEL" line up to the opening quote on the b64 blob
+	labelRegexp, _ := regexp.Compile(`LABEL.*(\")?com\.redhat\.apb\.spec(\")?\=(\\)?\n?\"`)
+	indices := labelRegexp.FindIndex(containerMeta)
+
+	if len(indices) == 0 {
+		log.Errorf("Didn't find expected bundle label in container metadata file")
+		return []byte{}
+	}
+	lineStartIndex := indices[0]
+	blobStartIndex := indices[1]
+	blobEndOffset := bytes.IndexByte(containerMeta[blobStartIndex:], byte('"'))
+	if blobEndOffset == -1 {
+		log.Errorf("Didn't find end of bundle label in container metadata file")
+		return []byte{}
+	}
+	blobEndIndex := blobStartIndex + blobEndOffset
+
+	// Build new label section
+	bundleMetaSection := containerMeta[lineStartIndex : blobStartIndex-1]
+	if !bytes.Contains(bundleMetaSection, []byte(lineBreakText)) {
+		bundleMetaSection = append(bundleMetaSection, lineBreakText...)
+	}
+	bundleMetaSection = append(bundleMetaSection, byte('"'))
+	bundleMetaSection = append(bundleMetaSection, bundleMetaEncoded...)
+
+	// Build final output with linebreaked "LABEL" b64 content
+	newContainerMeta := containerMeta[0:lineStartIndex]
+	newContainerMeta = append(newContainerMeta, []byte(bundleMetaSection)...)
```

Comment:
I don't think I need this `[]byte` cast anymore. Leftover from when I was using strings.